### PR TITLE
Banner component implementation, #2672

### LINKED ISSUE #2672: New Component: Banner
### Description  
The Banner is a mildly interruptive interface that provides contextual information and may only be dismissed by the user (unlike Snackbar and similarly to a Dialog but in a non-modal fashion). The Banner can contain templatable content and a collection of up to two actions.
[Material Banner](https://material.io/design/components/banners.html#)

simple

complex

### PR DESCRIPTION
**Pull request by wnvko**
Closes #2672

### CODE REVIEW

File: projects/igniteui-angular/src/lib/banner/banner.component.html [hunk 1]
```diff
@@ -0,0 +1,27 @@
+<igx-expansion-panel #expansionPanel [animationSettings]="animationSettings" (onCollapsed)="onExpansionPanelClose()" (onExpanded)="onExpansionPanelOpen()"
+    [collapsed]="collapsed">
+    <igx-expansion-panel-body>
+        <div class="igx-banner">
+            <div class="igx-banner__message">
+                <div *ngIf="bannerIcon" class="igx-banner__illustration">
```
Comment (ViktorSlavov) [inline projects/igniteui-angular/src/lib/banner/banner.component.html]: @desig9stein @simeonoff Could you take a look at the refactored template, when you get a chance?
Can the `igx-banner__illustration` class wrapper be removed? We don't need it to set a gray circle background for the image and that could simplify the template.

Comment (simeonoff) [reply projects/igniteui-angular/src/lib/banner/banner.component.html]: @ViktorSlavov Unfortunately the class is used to add other styles apart from setting a shape and background. @desig9stein Could you please remove the oval and the background?

**Comment by PlamenaMiteva** [comment]
Events in the sample do not fire because their names are wrong (should be onOpend/onClosed).

File: projects/igniteui-angular/src/lib/banner/banner.component.ts [hunk 2]
```diff
@@ -0,0 +1,165 @@
+import { Component, NgModule, EventEmitter, Output, Input, ViewChild, ElementRef, ContentChild } from '@angular/core';
+import { IgxExpansionPanelModule } from '../expansion-panel/expansion-panel.module';
+import { AnimationSettings } from '../expansion-panel/expansion-panel.component';
+import { BrowserModule } from '@angular/platform-browser';
```
Comment (rkaraivanov) [inline projects/igniteui-angular/src/lib/banner/banner.component.ts]: Don't import `BroweserModule` directly in our components. This is an application level import and will break the user's app when this module is lazy loaded for example.

Comment (ViktorSlavov) [reply projects/igniteui-angular/src/lib/banner/banner.component.ts]: Removed in the latest commit

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,8 @@ All notable changes for each version of this project will be documented in this
 
 ## 7.0.0
```
Comment (kdinev) [inline CHANGELOG.md]: This would be for `7.1.0`

Comment (ViktorSlavov) [reply CHANGELOG.md]: Resolved in the latest commit

File: projects/igniteui-angular/src/lib/banner/banner.component.ts [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+    event?: Event;
+}
+
+export interface BannerCancelEventArgs extends BannerEventArgs, CancelableEventArgs {
+}
+@Component({
+    selector: 'igx-banner',
+    templateUrl: 'banner.component.html'
+})
+export class IgxBannerComponent implements IToggleView {
```
Comment (damyanpetev) [inline projects/igniteui-angular/src/lib/banner/banner.component.ts]: @ViktorSlavov A top comment for entire component won't be bad, also small snippets on all the public properties.

File: projects/igniteui-angular/src/lib/expansion-panel/expansion-panel.component.ts [hunk 5]
```diff
@@ -17,6 +17,10 @@ import { IGX_EXPANSION_PANEL_COMPONENT, IgxExpansionPanelBase } from './expansio
 
 let NEXT_ID = 0;
 
+export interface AnimationSettings {
```
Comment (damyanpetev) [inline projects/igniteui-angular/src/lib/expansion-panel/expansion-panel.component.ts]: 👍

Comment (damyanpetev) [reply projects/igniteui-angular/src/lib/expansion-panel/expansion-panel.component.ts]: Could event be moved to core at some point.

File: projects/igniteui-angular/src/lib/expansion-panel/expansion-panel.component.ts [hunk 6]
```diff
@@ -121,9 +125,17 @@ export class IgxExpansionPanelComponent implements IgxExpansionPanelBase {
...
         if (this.collapsed) {
-            this.expand(evt);
+            this.open(evt);
         } else {
-            this.collapse(evt);
+            this.close(evt);
         }
     }
+
+    open(evt?: Event) {
```
Comment (damyanpetev) [inline projects/igniteui-angular/src/lib/expansion-panel/expansion-panel.component.ts]: Are those supposed to be a part of a different PR?

Comment (ViktorSlavov) [reply projects/igniteui-angular/src/lib/expansion-panel/expansion-panel.component.ts]: Yes, so that the expansion panel properly implements the IToggle interface.

File: projects/igniteui-angular/src/lib/expansion-panel/expansion-panel.component.ts (hunk 5 above)
Comment (ViktorSlavov) [reply projects/igniteui-angular/src/lib/expansion-panel/expansion-panel.component.ts]: Should I move it now? AFAIK the only components that expose animations of this type are the panel and the banner (using the panel), so maybe it's a bit unneeded

File: projects/igniteui-angular/src/lib/banner/banner.component.ts (hunk 4 above)
Comment (ViktorSlavov) [reply projects/igniteui-angular/src/lib/banner/banner.component.ts]: Thanks for pointing that out! I've added a top comment to the component. All of the public properties already had one (or were @hidden)

File: projects/igniteui-angular/src/lib/expansion-panel/expansion-panel.component.ts (hunk 5 above)
Comment (damyanpetev) [reply projects/igniteui-angular/src/lib/expansion-panel/expansion-panel.component.ts]: No need, this one is an improvement already :)
